lua/orgmode/org/mappings.lua

```diff
@@ -295,6 +295,31 @@ function OrgMappings:todo_prev_state()
   self:_todo_change_state('prev')
 end
 
+function OrgMappings:toggle_heading()
+  local line = vim.fn.getline('.')
+  local parent = Files.get_closest_headline()
+
+  if parent.line_number == vim.api.nvim_win_get_cursor(0)[1] then
+    line = line:gsub('^(%*+)', string.rep(' ', parent.level - 1))
+  else
+    line = line:gsub('^(%s*)', '')
+    if line:match('^[%*-]%s') then -- handle lists
+      line = line:gsub('^[%*-]%s', '') -- strip bullet
+      line = line:gsub('^%[([X%s])%]%s', function(checkbox_state)
+        if checkbox_state == 'X' then
+          return 'DONE '
+        else
+          return 'TODO '
```

Comment:
I'm now realizing that this does not take [org_todo_keywords](https://github.com/nvim-orgmode/orgmode/blob/master/DOCS.md#org_todo_keywords) into account. I will add this as well